network/validator/validator.go

```diff
@@ -3,14 +3,15 @@ package validator
 import (
 	"github.com/libp2p/go-libp2p/core/peer"
 
+	"github.com/onflow/flow-go/network/message"
 	"github.com/onflow/flow-go/network/p2p"
 )
 
 // MessageValidator validates the given message with original sender `from` and returns an error if validation fails
 // else upon successful validation it should return the decoded message type string.
 // Note: contrarily to pubsub.ValidatorEx, the peerID parameter does not represent the bearer of the message, but its source.
-type MessageValidator func(from peer.ID, msg interface{}) (string, error)
+type MessageValidator func(from peer.ID, msg *message.Message) (string, error)
```

Comment:
I've found no usage for the `MessageValidator` type. If we are not planning to use it, we may remove it.

Reply:
https://github.com/onflow/flow-go/pull/3831/commits/299d371c6b78472e077c28d82d4b90a385ec5954